<a href="https://colab.research.google.com/github/bangash-ds/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/bangash-ds/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

### **My lane is** Which pages should be reviewed first for a potential content refresh, based on observable search, content, and engagement signals?

### **Why:**
- This question focuses on identifying which pages need attention first by looking at measurable signals such as search performance, content quality or freshness, and user engagement. It helps prioritize pages where a content refresh could potentially improve visibility, traffic, and user interaction, rather than reviewing every page equally.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

### **What decision does your work improve?**

- Which pages should a reviewer prioritize for refresh review?

### **Who acts on it?**

- A content/SEO reviewer or strategist.

### **What does a wrong recommendation cost?**

- A wrong recommendation could waste time and resources on low-priority pages, while delaying improvements to pages that have stronger opportunities for better search visibility and engagement

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [7]:
import os
import sys
import subprocess

REPO_URL = "https://github.com/flyrank-bih/flyrank-ml-internship-starter"
REPO_DIR = "flyrank-ml-internship-starter"

if not os.path.isdir(REPO_DIR):
    subprocess.run(["git", "clone", "--depth", "1", REPO_URL, REPO_DIR], check=True)

os.chdir(REPO_DIR)

import pandas as pd
import numpy as np

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

df["is_declining_label"] = (
    df["trend_direction"].str.lower().eq("down").astype(int)
)

print("Dataset loaded:", df.shape)

Dataset loaded: (30000, 45)


In [8]:
print("Total pages:", len(df))

print("Declining pages:",
      round(df["is_declining_label"].mean() * 100, 2), "%")

print("Pages with 180+ days since last update:",
      round((df["days_since_last_update"] >= 180).mean() * 100, 2), "%")

Total pages: 30000
Declining pages: 54.21 %
Pages with 180+ days since last update: 0.58 %


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

- My analysis can identify pages that show observable signals associated with potential content-refresh opportunities, such as declining search performance, low CTR, high impressions, older content, or weak engagement. The results can be used as directional decision support to help prioritize which pages should be reviewed first. However, the analysis cannot prove that refreshing a page will cause better rankings, traffic, or engagement, and it cannot predict how Google will respond to a content change.

In [10]:
# Compare declining vs non-declining pages using observable signals

summary = df.groupby("is_declining_label")[
    ["impressions_90d", "content_age_days", "avg_position", "ctr", "engagement_rate"]
].mean().round(2)

summary

,impressions_90d,content_age_days,avg_position,ctr,engagement_rate
is_declining_label,,,,,
0,5533.31,279.83,16.82,0.73,2.65
1,4919.10,236.18,15.94,0.32,2.44


### **What I can claim:**
- The data shows observable differences between declining and non-declining pages. For example, declining pages have lower average CTR (0.32 vs. 0.73) and lower impressions over 90 days (4,919 vs. 5,533). These signals can be used to prioritize pages for content review and provide directional decision support.

### **What I cannot claim:**
- These results show associations, not causation. I cannot claim that refreshing a page will cause its CTR or traffic to increase, or that the analysis can predict Google's future rankings.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.